src/lib/misc.c

```diff
@@ -1153,8 +1153,12 @@ grabdate(const char *s, int64_t *t)
 
     if (!compiled) {
         compiled = 1;
-        (void) regcomp(
-          &r, "([0-9][0-9][0-9][0-9])[-/]([0-9][0-9])[-/]([0-9][0-9])", REG_EXTENDED);
+        if (regcomp(&r,
+                    "([0-9][0-9][0-9][0-9])[-/]([0-9][0-9])[-/]([0-9][0-9])",
+                    REG_EXTENDED) != 0) {
+            fprintf(stderr, "Can't compile regex\n");
```

Comment:
I'm not sure because this may happened when rnp hasn't got enough memory or regex is wrong. This is hardcoded regex and it shouldn't be wrong (or we should get it by tests). So, the error means that we haven't enought memory or something like this.

Are you sure that it is a good idea use `pgp_error` for this?